hail/python/hailtop/batch/backend.py

```diff
@@ -51,6 +51,24 @@ class Backend(abc.ABC, Generic[RunningBatchType]):
 
     _closed = False
 
+    def __init__(self):
+        self._batch_fses: Dict["batch.Batch", "RouterAsyncFS"] = {}
+
+    def batch_fs(self, batch: "batch.Batch") -> "RouterAsyncFS":
+        try:
+            return self._batch_fses[batch]
+        except KeyError:
+            if batch.requester_pays_project is not None:
+                self._batch_fses[batch] = RouterAsyncFS(
+                    gcs_kwargs={"gcs_requester_pays_configuration": batch.requester_pays_project}
```

Comment:
If we're now storing a separate FS per batch, why does the requester pays configuration on the FS need to be overridable? Can we remove the change to override that in `GoogleStorageClient`?

Reply:
good catch, removed the arg that could override the storage client's requester pays project for `bucket_info`